hivemind/client/optim/simple.py

```diff
@@ -0,0 +1,88 @@
+from threading import Thread, Lock, Event
+from typing import Optional, Callable, Any
+
+import torch
+
+from hivemind.dht import DHT
+from hivemind.client.averaging import DecentralizedAverager
+from hivemind.client.optim.base import DecentralizedOptimizerBase
+from hivemind.utils import MPFuture, get_logger
+
+logger = get_logger(__name__)
+
+
+class DecentralizedOptimizer(DecentralizedOptimizerBase):
```

Comment:
`ParameterAveragingOptimizer`? (just to clarify that this optimizer only does parameter averaging)

Reply:
Alternatively, we can have one DecentralizedOptimizer class which averages parameters,gradients or statistics depending on use case. It shouldn't be too large, given that most of the work is done in DecentralizedAverager

Reply:
Renamed (first option)

While most of the work is indeed handled by the DecentralizedAverager, i am unsure which user inter would be preferable for the generalized optimizer (with local gradients, statistics, etc).

Let us handle that in subsequent pull requests.

Reply:
Also: added a DecentralizedSGD ([term usage example](https://arxiv.org/pdf/2003.10422.pdf)) wrapper for convenience -- mainly not to scare people off with ParameterAveragingOptimizer. To the best of my knowledge, SGD is the only mainstream optimizer from torch.optim that is guaranteed to converge with parameter averaging.

Note: i intend to use DecentralizedSGD for resnet50 example after some performance tweaks.